pkg/common/aws/iam.go

```diff
@@ -0,0 +1,209 @@
+package aws
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/iam"
+)
+
+var (
+	rolesubstr     = "osde2e"
+	providersubstr = "cloudfront"
+)
+
+func (CcsAwsSession *ccsAwsSession) CleanupOpenIDConnectProviders(olderthan time.Duration, dryrun bool) error {
+	err := CcsAwsSession.GetAWSSessions()
+	if err != nil {
+		return err
+	}
+
+	input := &iam.ListOpenIDConnectProvidersInput{}
+	result, err := CcsAwsSession.iam.ListOpenIDConnectProviders(input)
+	if err != nil {
+		return err
+	}
+
+	for _, provider := range result.OpenIDConnectProviderList {
+		// Create GetOpenIDConnectProviderInput struct
+		input := &iam.GetOpenIDConnectProviderInput{
+			OpenIDConnectProviderArn: provider.Arn,
+		}
+
+		// Get the provider
+		result, err := CcsAwsSession.iam.GetOpenIDConnectProvider(input)
+		if err != nil {
+			return err
+		}
+
+		// If provider name contains "cloudfront" and is older than given days, delete it
+		if strings.Contains(*result.Url, providersubstr) && time.Since(*result.CreateDate) > olderthan {
+			fmt.Printf("Provider will be deleted: %s\n", *provider.Arn)
+
+			if !dryrun {
+				// Create DeleteOpenIDConnectProviderInput struct
+				input := &iam.DeleteOpenIDConnectProviderInput{
+					OpenIDConnectProviderArn: provider.Arn,
+				}
+				_, err := CcsAwsSession.iam.DeleteOpenIDConnectProvider(input)
+				if err != nil {
+					return err
+				}
+				fmt.Println("Deleted")
+			}
+		}
+	}
+
+	return nil
+}
+
+func (CcsAwsSession *ccsAwsSession) CleanupRoles(olderthan time.Duration, dryrun bool) error {
+	err := CcsAwsSession.GetAWSSessions()
+	if err != nil {
+		return err
+	}
+
+	input := &iam.ListRolesInput{
+		MaxItems: aws.Int64(1000),
+	}
+	result, err := CcsAwsSession.iam.ListRoles(input)
+	if err != nil {
+		return err
+	}
+
+	for _, role := range result.Roles {
+		if strings.Contains(*role.Arn, rolesubstr) && time.Since(*role.CreateDate) > olderthan {
```

Comment:
same for this, I would like to understand why we want to leave resources up for longer than the cluster is running